src/core.c/Kernel.pm6

```diff
@@ -164,11 +164,15 @@ class Kernel does Systemic {
     multi method signal(Kernel:D: Signal:D \signal --> Int:D) { signal.value }
     multi method signal(Kernel:D: Int:D    \signal --> Int:D) { signal       }
 
-    my int $cpu-cores = nqp::cpucores;
-    my int $cpu-cores-but-one = max $cpu-cores - 1, 1;
+    method cpu-cores(--> Int) { nqp::cpucores }
 
-    method cpu-cores(--> Int) { $cpu-cores }
-    method cpu-cores-but-one() is implementation-detail { $cpu-cores-but-one }
+    my $cpu-cores-but-one := nqp::null;
+    method cpu-cores-but-one() is implementation-detail {
```

Comment:
I feel a little inconsistency between .cpu-cores and .cpu-cores-but-one (the first returns dynamic value, the second - static value). implementation-detail trait forgives it of course. What do you think about:
```
method cpu-cores(--> Int) {
  my $cpu-cores := nqp::cpucores;
  $cpu-cores-but-one := max $cpu-cores - 1, 1;
  $cpu-cores
}
```

Reply:
Or leave only one method
```
my int $cpu-cores := nqp::null;
method cpu-cores(:$last-value = False --> Int) {
  nqp::ifnull(
    $cpu-cores,
    $cpu-cores := nqp::cpucores()
  )
  nqp::if(
    $last-value,
    $cpu-cores,
    $cpu-cores := nqp::cpucores()
  )
}
[...]
max Kernel.cpu-cores(:last-value)-1, 1
```

Reply:
The current API of `Kernel.cpu-cores` is that it will always look up the value.  It was felt that in some situations, making it return a static value, would be incorrect.

On the other hand, it turned out that `nqp::cpucores` is way more expensive than was thought **and** it is causing issues when threading is involved.  So, on the one hand we want to reduce the CPU used, and on the other hand we only want to call `nqp::cpucores` when it is actually necessary.  This is now what the internal `Kernel.cpu-cores-but-one` does, as that is the default value we need to have in all cases internally.

Reply:
Oh, I missed this moment that `Kernel.cpu-cores` is public API, sorry. I'm fully understand why you introduced `Kernel.cpu-cores-but-one`. Probably it is only naming related inconsistency in my head. Nevermind. Thank you for your time.